homeassistant/components/ai_speaker/config_flow.py

```diff
@@ -1,65 +1,87 @@
+from .const import DOMAIN  # pylint:disable=unused-import
 
 _LOGGER = logging.getLogger(__name__)
-AIS_CONFIG = {
-    vol.Required(CONF_HOST): str,
-}
 
+STEP_USER_DATA_SCHEMA = vol.Schema({"host": str})
 
-class AisConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
-    """Handle a AI-Speaker config flow."""
+
+class AisDevice:
+    """Ais device class."""
+
+    def __init__(self, hass, host):
+        """Initialize."""
+        self.hass = hass
+        self.host = host
+        self.web_session = aiohttp_client.async_get_clientsession(self.hass)
+
+    async def get_gate_info(self):
+        """Return the ais gate info."""
+        ais_ws = AisWebService(self.hass.loop, self.web_session, self.host)
+        return await ais_ws.get_gate_info()
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    ais_gate = AisDevice(hass, data["host"])
+
+    ais_gate_info = await ais_gate.get_gate_info()
+    if ais_gate_info is None:
+        raise InvalidAuth
+
+    product = ais_gate_info["Product"]
+    ais_id = ais_gate_info["ais_id"]
+    return {"title": f"AI-Speaker {product}.", "ais_id": ais_id}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for AI Speaker."""
 
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
-    DOMAIN = DOMAIN
 
     async def async_step_user(self, user_input=None):
-        """Commissioning the configuration by the user."""
-        return self.async_show_form(step_id="confirm")
-
-    async def async_step_confirm(self, user_input=None):
-        """User confirmation step."""
-        if user_input is not None:
+        """Handle the initial step."""
+        if user_input is None:
             return self.async_show_form(
-                step_id="settings", data_schema=vol.Schema(AIS_CONFIG)
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
             )
 
-        return self.async_show_form(step_id="confirm")
-
-    async def async_step_settings(self, user_input=None):
-        """Step of the connection settings."""
-        if user_input is not None:
-            web_session = aiohttp_client.async_get_clientsession(self.hass)
-            ais_host = user_input["host"]
-            ais_gate = AisWebService(self.hass.loop, web_session, ais_host)
-            ais_gate_info = await ais_gate.get_gate_info()
-            ais_id = ais_gate_info.get("ais_id")
-            if ais_id is None:
-                errors = {CONF_HOST: "discovery_error"}
-                return self.async_show_form(
-                    step_id="settings",
-                    data_schema=vol.Schema(AIS_CONFIG),
-                    errors=errors,
-                )
+        errors = {}
 
+        try:
+            info = await validate_input(self.hass, user_input)
+        except CannotConnect:
```

Comment:
We're only raising `InvalidAuth`.
```suggestion
        except InvalidAuth:
```